Mage.Sets/src/mage/cards/e/EpicExperiment.java

```diff
@@ -61,59 +62,16 @@ public EpicExperimentEffect(final EpicExperimentEffect effect) {
     public boolean apply(Game game, Ability source) {
         Player controller = game.getPlayer(source.getControllerId());
         MageObject sourceObject = source.getSourceObject(game);
-        if (controller != null && sourceObject != null) {
-            // move cards from library to exile
-            controller.moveCardsToExile(controller.getLibrary().getTopCards(game,
-                    source.getManaCostsToPay().getX()), source, game, true,
-                    source.getSourceId(), sourceObject.getIdName());
-            // cast the possible cards without paying the mana
-            ExileZone epicExperimentExileZone = game.getExile().getExileZone(source.getSourceId());
```

Comment:
I added another check but I'm not sure that it's necessary

Reply:
Some replacement effects can change destination zone, so card can be in non exile zone at the end. It's a popular bug.

![shot_210818_164834](https://user-images.githubusercontent.com/8344157/129900875-68bd1a3e-55fb-49d7-967c-cc39a6002632.png)

Reply:
yeah I was thinking of exile in particular but you’re right, also it gets annoying when the card doesn’t state where it can cart from so it still lets you cast from the new zone